packages/web/src/app/actions/requestLessonGeneration.ts

```
@@ -14,11 +15,77 @@ interface RequestGenerationResult {
+			return {
+				success: false,
+				error: 'يتم معالجة هذا الدرس حالياً',
+				redirectUrl: `/request/status/${youtubeId}`,
+			};
+			
+		case 'failed':
+		case 'cancelled':
+			// Check retry cooldown (1 hour)
+			const oneHourAgo = new Date(Date.now() - 60 * 60 * 1000);
+			if (job.updated_at > oneHourAgo) {
+				const waitMinutes = Math.ceil((60 * 60 * 1000 - (Date.now() - job.updated_at.getTime())) / (1000 * 60));
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)